yarn_api_client/base.py

```diff
@@ -6,43 +6,62 @@
 
 from .errors import APIError, ConfigurationError
 
+try:
+    from urlparse import urlparse, urlunparse
+except ImportError:
+    from urllib.parse import urlparse, urlunparse
 
 class Response(object):
     def __init__(self, response):
         self.data = response.json()
 
-
 class BaseYarnAPI(object):
     __logger = None
     response_class = Response
 
-    def __init__(self, address=None, port=None, timeout=None, kerberos_enabled=None):
-        self.address, self.port, self.timeout, self.kerberos_enabled = address, port, timeout, kerberos_enabled
+    def __init__(self, service_endpoint=None, timeout=None, auth=None, verify=True):
+        self.timeout = timeout
+
+        if service_endpoint:
+            # Snippet to decouple parts of Uri for further building of URLs
+            self.service_uri = urlparse(service_endpoint)
+            self.service_uri_scheme = self.service_uri.scheme or 'http'
+            self.service_uri_hostname = self.service_uri.hostname or self.service_uri.path
+
+            if self.service_uri.port:
+                self.service_uri_postfix = ':' + str(self.service_uri.port)
+            else:
+                self.service_uri_postfix = ''
+        else:
+            self.service_uri = None
+            self.service_uri_scheme = None
+            self.service_uri_hostname = None
+            self.service_uri_postfix = None
```

Comment:
Can we please follow @lresende's lead and create a local `Uri` class that encapsulates this?  I don't see the need for all the extra member variables.  This class would also be used by the test code and eliminates the `import` hassle.  The class can be defined in `base.py` or a new file.

The constructor should call `urlparse` and initialize default values - as Luciano's previous snippet laid out.

Please keep `port` separated from the `:` and add that in when constructing the url.  The class can have a `to_url(api_path)` method that constructs a url.

Why is the `path` used as the `service_uri_hostname` if no `hostname` is parsed?  That doesn't seem correct.

Reply:
If user passes domain without scheme, then all lands inside path. 
Example - pass service_endpoint `localhost`

Reply:
Interesting about hostname and path.  Thanks for the explanation.

While I agree its a thin wrapper, it does provide some encapsulation so we can make changes in the future.  Thanks.

Reply:
I tried to incorporate best of two ideas, check latest commit.